src/ride/track.c

```diff
@@ -4880,15 +4881,587 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+        // ---- loc_6CD9F0:
+        // mov     al, [esi]
+        uint8 temp_al = temp_esi_map->type;
+
+        // and     al, 3Ch
+        temp_al = temp_al & 0x3C;
+
+        // cmp     al, 8
+        // jnz     short loc_6CDA0C
+        if (temp_al == 8) {
+            // cmp     dl, [esi+2]
+            // jnz     short loc_6CDA0C
+            if ((temp_dx & 0xff) == temp_esi_map->base_height) {
+                // cmp     bl, [esi+7]
+                // jnz     short loc_6CDA0C
+                if (rideIndex == temp_esi_map->properties.surface.ownership) {
+                    // cmp     byte ptr [esi+4], 65h ; 'e'
+                    // jz      loc_6CDB35
+                    if (temp_esi_map->properties.surface.slope == 0x65) {
+                        return loc_6CDB35();
+                    }
+                }
+            }
+        }
+
+        // ---- loc_6CDA0C:
+        // add     esi, 8
+        // test    byte ptr [esi-7], 80h
+        // jz      short loc_6CD9F0
+        temp_esi += 8;
+        if (RCT2_GLOBAL(temp_esi - 7, uint8) != 0x80) {
+            break;
+        }
+    }
+
+
+    // mov     edx, [esp+14h+var_10]
+    uint32 temp_edx = RCT2_GLOBAL(esp + 0x14 + 0x10, uint32);
+
+    // or      dh, dh
+    // jnz     loc_6CDE43
+    if ((temp_edx & 0xFF00) != 0) {
+        return loc_6CDE43();
+    }
+
+    // mov     eax, [esp+14h+var_4]
+    eax = eax;
+
+    // mov     ecx, [esp+14h+var_C]
+    ecx = ecx;
+
+    // and     ax, 0FFE0h
+    uint16 x_floored = (uint16) floor2(x, 32);
+
+    // and     cx, 0FFE0h
+    uint16 y_floored = (uint16) floor2(y, 32);
+
+    // mov     edx, [esp+14h+var_14]
+    // shr     dx, 3
+    uint16 tmp_dx = z >> 3;
+
+    // mov     dh, dl
+    tmp_dx = (tmp_dx & 0xFF) + ((tmp_dx & 0xFF) << 8);
+
+    // add     dh, 4
+    tmp_dx += 0x4 << 8;
+
+    // mov     bl, 0Fh
+    // call    can_construct_at
+    // TODO: find out what these somethings are
+    // jb      loc_6CDE4C
+    if (!map_can_construct_at(x_floored, y_floored, something, something_else, 0x0F)) {
+        // TODO: should this be inverted?
+        loc_6CDE4C();
+    }
+
+    // test    byte_F1AD60, 4
+    // jnz     loc_6CDE22
+    if (RCT2_GLOBAL(RCT2_ADDRESS_ELEMENT_LOCATION_COMPARED_TO_GROUND_AND_WATER, uint8) != 0x04) {
+        return loc_6CDE22();
+    }
+
+    // test    byte_F1AD60, 2
+    // jnz     loc_6CDE17
+    if (RCT2_GLOBAL(RCT2_ADDRESS_ELEMENT_LOCATION_COMPARED_TO_GROUND_AND_WATER, uint8) != 0x02) {
+        return loc_6CDE17();
+    }
+
+    // movzx   esi, byte ptr [esp+14h+var_10]
+    esi = RCT2_GLOBAL(edx, uint8);
+
+    // imul    esi, 260h
+    esi = esi * 0x260;
+
+    // push    eax
+    // push    edx
+    // push    edi
+    
+    // movzx   edi, rides[esi]
+    rct_ride *ride = GET_RIDE(esi);
+    
+    // movzx   eax, ds:word_97DD78[edi*4]
+    // imul    ds:dword_99DBC8
+    money32 price = RCT2_ADDRESS(0x0097DD78, money16)[ride->type * 4] * RCT2_GLOBAL(0x0099DBC8, money32);
```

Comment:
Remember this is C code not assembly ride->type * 4 will be timesed by 2 due to accessing words so really this is now `movzx   eax, ds:word_97DD78[edi*8]` which is too much.